From e83b909a713bc5ac14fd33a00ccd370bff2f18a0 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Fri, 17 Dec 2021 11:19:15 +0800
Subject: [PATCH 1/2] intel: fcs: release mutex lock in failing path

commit 00295d5977edb60301c2aace0391bee6f35d34a3 upstream

Signed-off-by: Meng Li <Meng.Li@windriver.com>
[DP: backport of 00295d5977ed from v5.10/standard/intel-sdk-5.10/intel-socfpga branch of linux-yocto]
Signed-off-by: Dragos-Marian Panait <dragos.panait@windriver.com>
---
 drivers/crypto/intel_fcs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/intel_fcs.c b/drivers/crypto/intel_fcs.c
index 1b193d9d966b..a8c8d8f2a99a 100644
--- a/drivers/crypto/intel_fcs.c
+++ b/drivers/crypto/intel_fcs.c
@@ -218,8 +218,10 @@ static int fcs_request_service(struct intel_fcs_priv *priv,
 	reinit_completion(&priv->completion);
 
 	ret = stratix10_svc_send(priv->chan, p_msg);
-	if (ret)
+	if (ret) {
+		mutex_unlock(&priv->lock);
 		return -EINVAL;
+	}
 
 	ret = wait_for_completion_timeout(&priv->completion,
 							timeout);
-- 
2.34.1

